src/Gui/InputField.cpp

```diff
@@ -86,22 +86,33 @@ InputField::InputField(QWidget * parent)
     }
     iconLabel = new ExpressionLabel(this);
     iconLabel->setCursor(Qt::ArrowCursor);
-    QPixmap pixmap = getValidationIcon(":/icons/button_valid.svg", QSize(sizeHint().height(),sizeHint().height()));
+    QFontMetrics fm(font());
+    int iconSize = fm.height();
+    QPixmap pixmap = getValidationIcon(":/icons/button_valid.svg", QSize(iconSize, iconSize));
     iconLabel->setPixmap(pixmap);
-    iconLabel->setStyleSheet(QStringLiteral("QLabel { border: none; padding: 0px; }"));
     iconLabel->hide();
     connect(this, &QLineEdit::textChanged, this, &InputField::updateIconLabel);
-    int frameWidth = style()->pixelMetric(QStyle::PM_DefaultFrameWidth);
-    setStyleSheet(QStringLiteral("QLineEdit { padding-right: %1px } ").arg(iconLabel->sizeHint().width() + frameWidth + 1));
-    QSize msz = minimumSizeHint();
-    setMinimumSize(qMax(msz.width(), iconLabel->sizeHint().height() + frameWidth * 2 + 2),
-                   qMax(msz.height(), iconLabel->sizeHint().height() + frameWidth * 2 + 2));
+
+    // Set Margins
+    // vertical to avoid this: https://forum.freecad.org/viewtopic.php?f=8&t=50615
+    // horizontal to avoid going under the icon

```

Comment:
These comments make no sense. I have no idea how the forum thread is realted either.

Reply:
```suggestion
    // vertical margin, such that `,` won't clipped to a `.` and similar font descents. Relevant on some OSX versions
    // horizontal margin, such that text will not be behind `fx` icon
```